feat: support for pointers to primitives in Scan() and HSet()

numberinho opened the pull request:
Easy way to implement support for pointers to primitives at the `Scan()` method. Uses built-in decoders by just passing the kind of the element of the pointer to the decoders map.

rueian commented:
Hi @numberinho, thank you for your contribution!

These changes look good to me. Could you also modify the testes here https://github.com/redis/rueidis/blob/main/rueidiscompat/hscan_test.go to cover these changes?

numberinho commented:
Hey @rueian, thank you for this awesome library and your super fast replies! I really appreciate your effort and work :) 

I tried adding a test, but I'm not very used to the testing framework. Hopefully I implemented it in the right way.

Did you notice that I also added some lines to modify the `HSet()` - functioning to omit setting Go nil pointers to Redis in [rueidiscompat/adapter.go](https://app.codecov.io/gh/redis/rueidis/pull/388?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=redis#diff-cnVlaWRpc2NvbXBhdC9hZGFwdGVyLmdv) ? How does this look to you?

rueian commented:
> I tried adding a test, but I'm not very used to the testing framework. Hopefully I implemented it in the right way.

It is in the right way. The test is passed.

> Did you notice that I also added some lines to modify the `HSet()` - functioning to omit setting Go nil pointers to Redis in [rueidiscompat/adapter.go](https://app.codecov.io/gh/redis/rueidis/pull/388?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=redis#diff-cnVlaWRpc2NvbXBhdC9hZGFwdGVyLmdv) ? How does this look to you?

They also look good to me, but I notice that the test does not cover them.

Could you also edit another test here to cover them?
https://github.com/redis/rueidis/blob/f0f37fc32e7ef78bf169960a24930c2709070a3f/rueidiscompat/adapter_test.go#L2242-L2296
